source/fundamentals/poco.txt

```diff
@@ -0,0 +1,249 @@
+
+      .. code-block:: json
+         :copyable: false
+
+         {
+           "_id": ObjectId("..."),
+           "name": "Long Sleeve Shirt",
+           "instock": true,
+           "price": 17.99,
+           "color": [ "black", "navy", "red" ]
+         }
+
+You can define a POCO with any object structure that suits your
+needs, including nested objects, arrays, lists, and any data types. The
+driver serializes POCOs using default mapping rules to generate the
+appropriate BSON representation of the document.
+
+Custom Serialization
+--------------------
+
+If the default field mapping behavior does not meet your needs, you can
+specify custom behavior using serialization related attributes. There
+are various attributes which you can specify to modify the way that the
+driver serializes each property of your POCO. This section describes
+some of the common serialization related attributes.
+
+Set Field Names
+~~~~~~~~~~~~~~~
+
+The driver serializes POCO properties to BSON fields with same field
+name and capitalization. To store a property under a different name, you
+can use the ``BsonElementAttribute``. The following code shows how you
+can direct the driver to store the ``YearBuilt`` property of the
+``House`` object as ``year_built`` in the serialized BSON document:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonElement("year_built")]
+       public int YearBuilt { get; set; }
+   }
+
+Select Type Representation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For some {+language+} primitive types you can control the BSON type used
+to represent the value of your POCO property using the
+``BsonRepresentationAttribute``. The following code shows how you can
+specify that the ``YearBuilt`` property, defined as a ``char`` in
+{+language+}, is serialized as a BSON ``Int32`` type:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonRepresentation(BsonType.Int32)]
+       public char YearBuilt { get; set; }
+   }
+
+Set Field Order
+~~~~~~~~~~~~~~~
+
+The driver serializes properties to BSON fields in the order they
+are specified in the POCO. To store properties in a custom order, you
+can specify the ``Order`` named parameter in the ``BsonElement``
+attribute. The following code shows how you can direct the driver to
+store the ``YearBuilt`` property after the ``Style`` property:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonElement(Order = 2)]
+       public int YearBuilt { get; set; }
+
+       [BsonElement(Order = 1)]
+       public string Style { get; set; }
+   }
+
+Any properties that do not have an explicit ``Order`` will occur after
+those that do.
+
+Identify Id Property
+~~~~~~~~~~~~~~~~~~~~
+
+By default, the driver uses any public property named ``Id``, ``id``, or ``_id`` as the
```

Comment:
Issue: Do we know what happens if I have more than one such property?

Suggestion: Link or explanation, possibly a warning box if it's something we want users to avoid.